scrapy/core/engine.py

```diff
@@ -311,39 +293,24 @@ def _schedule_request(self, request: Request, spider: Spider) -> None:
                 signals.request_dropped, request=request, spider=spider
             )
 
-    def download(self, request: Request, spider: Optional[Spider] = None) -> Deferred:
+    def download(self, request: Request) -> Deferred:
         """Return a Deferred which fires with a Response as result, only downloader middlewares are applied"""
-        if spider is not None:
-            warnings.warn(
-                "Passing a 'spider' argument to ExecutionEngine.download is deprecated",
-                category=ScrapyDeprecationWarning,
-                stacklevel=2,
-            )
-            if spider is not self.spider:
-                logger.warning(
-                    "The spider '%s' does not match the open spider", spider.name
-                )
         if self.spider is None:
             raise RuntimeError(f"No open spider to crawl: {request}")
-        return self._download(request, spider).addBoth(
-            self._downloaded, request, spider
-        )
+        return self._download(request).addBoth(self._downloaded, request)
```

Comment:
Here's the culprit from the previous attempt. I forgot tho remove spider parameter from `.addBoth` call after removing from `_downloaded` method.